### PR TITLE
CMake DDR: Fix bug where #undef statements were not recognized

### PR DESCRIPTION
**Pull request by dnakamura**
Signed-off-by: Devin Nakamura <devinn@ca.ibm.com>

### CODE REVIEW

**Comment by charliegracie** [comment]
@genie-omr build all

**Comment by charliegracie** [comment]
@keithc-ca does this look logically correct to you?

File: cmake/modules/ddr/cmake_ddr.awk [hunk 1]
```diff
@@ -142,12 +142,13 @@ NR == 1 {
 {
 	# replace tabs with spaces, and cut multiple spaces
 	sub(/[ \t]+/, " ")
```
Comment (keithc-ca) [inline cmake/modules/ddr/cmake_ddr.awk]: This needs to use `gsub` to fix all places with two or more whitespace characters.

File: cmake/modules/ddr/cmake_ddr.awk [hunk 2]
```diff
@@ -142,12 +142,13 @@ NR == 1 {
...
-	#clean up any space between '#' and define
+	# clean up any space between '#' and define
 	sub(/^ ?# +define/, "#define")
```
Comment (keithc-ca) [inline cmake/modules/ddr/cmake_ddr.awk]: Why `+` here sand not `?`?

File: cmake/modules/ddr/cmake_ddr.awk [hunk 3]
```diff
@@ -142,12 +142,13 @@ NR == 1 {
...
-
+	# Normalize #undefs, ie remove excess spaces, and possible leading '/*'
+	sub(/^ ?(\/\* ?)?# ?undef/, "#undef")
 }
 
-/^# ?define / {
+/^#define / {
 	sub(/^ ?# ?define /, "#define ")
```
Comment (keithc-ca) [inline cmake/modules/ddr/cmake_ddr.awk]: This use of `sub` will become useless.

**Comment by dnakamura** [comment]
@charliegracie  fixed all issues

**Comment by charliegracie** [comment]
@genie-omr build all

**Pull request merged**